packages/matchbox/src/components/Box/Box.js

```diff
@@ -0,0 +1,47 @@
+import styled from 'styled-components';
+import { border, color, flexbox, grid, layout, position, shadow, space, typography, compose } from 'styled-system';
+import propTypes from '@styled-system/prop-types';
+import PropTypes from 'prop-types';
+
+const system = compose(
+  border,
+  color,
+  flexbox,
+  grid,
+  layout,
+  position,
+  shadow,
+  space,
+  typography
+);
+
+const truncate = (props) => {
+  if (props.truncate) {
+    return {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      whiteSpace: 'nowrap'
+    };
+  }
+};
+
+const Box = styled('div')`
+  ${system}
+  ${truncate}
+`;
+
+Box.propTypes = {
+  ...propTypes.border,
+  ...propTypes.color,
+  ...propTypes.flexbox,
+  ...propTypes.grid,
+  ...propTypes.layout,
+  ...propTypes.position,
+  ...propTypes.shadow,
+  ...propTypes.space,
+  ...propTypes.typography,
+  truncate: PropTypes.bool
+};
+
+Box.displayName = 'Box';
+export default Box;
```

Comment:
Can arbitrary props be passed in to this component, i.e., data attributes or `aria-label` or things like that?

Reply:
Yep, I'll add a test to make sure valid attributes get passed through